presto-main/src/main/java/io/prestosql/server/PrestoServer.java

```diff
@@ -126,12 +117,7 @@ public void run()
 
             injector.getInstance(StaticCatalogStore.class).loadCatalogs();
 
-            // TODO: remove this huge hack
-            updateConnectorIds(
-                    injector.getInstance(Announcer.class),
-                    injector.getInstance(CatalogManager.class),
-                    injector.getInstance(ServerConfig.class),
-                    injector.getInstance(NodeSchedulerConfig.class));
+            injector.getInstance(ConnectorManager.class).updateConnectorIds();
```

Comment:
We can push this logic to the constructor or maybe add an annotation `PostConstructor` to the `ConnectorManager#updateConnectorIds` so that it would be invoked after `ConnectorManager` is created.

Reply:
@Praveen2112 Seems like we can announce connector Id if we constructed all connector from plugin
```
injector.getInstance(PluginManager.class).loadPlugins();
injector.getInstance(StaticCatalogStore.class).loadCatalogs();
```
not sure whether we could update immediate after ` ConnectorManager ` constructed ...